08_education/college_metrics_county.qmd

```diff
@@ -0,0 +1,1171 @@
+
+```{r}
+count(repwts_person, sample)
+```
+
+Remove the sample variable.
+```{r}
+repwts_person <- repwts_person %>% 
+  select(-sample)
+```
+
+### Merge on repwts
+
+```{r}
+length(unique(pull(acs, unique_person_id)))
+length(unique(pull(repwts_person, unique_person_id)))
+acs_combined <- 
+  left_join(
+    acs,
+    repwts_person,
+    by = "unique_person_id"
+  )
+length(unique(pull(acs_combined, unique_person_id)))
+rm(acs)
+rm(repwts_person)
+```
+
+Check that the merge did not result in any missing values.
+```{r}
+stopifnot(all(map_dbl(acs_combined, ~sum(is.na(.x))) == 0))
+```
+
+
+## Clean Data
+
+Missing data is reported in a variety of ways by IPUMS data. This step walks through the missing values in key variables and checks that we are dealing with them appropriately. 
+
+Age: No missing values reported in variable description. Check the distribution to confirm there are no abnormalities: the distribution of age looks reasonable (no bunching or extreme outliers).
+
+```{r}
+acs_combined %>% 
+  ggplot(aes(x = age)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Age Variable Results") +
+  ylab("Density")
+```
+
+EDUCD: N/As reported as "1" in all modern surveys. N/As were "000" in the 1980 and 1950 decennial census surveys (not used here).
+Variable codes explained [here]([https://usa.ipums.org/usa-action/variables/EDUC#codes_section) & see "Detailed codes" at bottom).
+Vacancy is reported as "0". 
+
+There is a significant share of records reported as "1" for the education variable. 
+
+```{r}
+acs_combined %>% 
+  mutate(educd = zap_labels(educd)) %>% 
+  count(educd) %>% 
+  ggplot(mapping = aes(x = factor(educd), y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Educd, All Records") +
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Educational Attainment",
+       y = NULL)
+```
+
+Confirm the missing values from the value labels of the educd variable to ensure we are eliminating the correct values. "0", "1" and "999" are N/A or missing and should be turned into NA values that R recognizes. 
+
+Note that there are no records with values of zero in the bar chart above which eliminates the need to deal with the "no schooling" option.
+```{r}
+head(ipums_val_labels(acs_combined$educd), 50) %>% 
+  reactable()
+```
+
+Turn the missing/"not in universe" value from the educd variable into NAs and remove the labels for the rest of the analysis. Rename to the more meaningful "education_detailed". 
+```{r}
+acs_clean <- acs_combined %>%
+  mutate(education_detailed = zap_labels(educd), 
+         education_detailed = ifelse(education_detailed %in% c(1, 0, 999), NA_integer_,
+                          education_detailed)
+  )
+```
+
+Look at the distribution of education_detailed after adjustment for NAs. We can see that there are no longer records for "1" but we now see "NA".
```

Comment:
Reference comments above; there are no "NA" values at this point because there were not missing categories to begin with.